src/components/Visualization/useAnalyticsData.js

```diff
@@ -35,22 +43,66 @@ const formatRowValue = (rowValue, header, metaDataItems) => {
     }
 }
 
-const extractHeadersFromColumns = (columns) =>
-    columns.reduce((headers, { dimension }) => {
-        switch (dimension) {
-            // TODO remove when this is sorted out https://jira.dhis2.org/browse/TECH-869
-            case 'pe':
-                break
-            case 'ou':
-                headers.push('ouname')
-                break
-            default:
-                headers.push(dimension)
-                break
-        }
+const isTimeDimension = (dimensionId) =>
+    [
+        DIMENSION_TYPE_EVENT_DATE,
+        DIMENSION_TYPE_ENROLLMENT_DATE,
+        DIMENSION_TYPE_INCIDENT_DATE,
+        DIMENSION_TYPE_SCHEDULED_DATE,
+        DIMENSION_TYPE_LAST_UPDATED,
+    ].includes(dimensionId)
+
+const getAdaptedVisualization = (visualization) => {
+    const adaptedColumns = []
+    const adaptedRows = []
+    const adaptedFilters = []
+    const headers = []
+    const timeDimensionParameters = {}
+
+    visualization.columns.forEach((dimensionObj) => {
+        const dimensionId = dimensionObj.dimension
```

Comment:
since this is repeated thrice with the only difference being which array its pushing to in the end, maybe its worth extracting this to a generic function that's called for each axis with an input array to know where to push the end result? (e.g. `adaptedColumns.push(dimensionObj)` is replaced by `outputArray.push(dimensionObj)`)

Reply:
Refactored.